src/Toolbox.php

```diff
@@ -3271,4 +3271,108 @@ final public static function getNormalizedItemtype(string $itemtype)
+        }
+        return true;
+    }
+
+    /**
+     * Get an element within the array at the given path.
+     * @param array $array The array to check
+     * @param string $path The path to the element
+     * @param string $path_delimiter The delimiter used in the path
+     * @return mixed
+     */
+    public static function getElementByArrayPath(array $array, string $path, string $path_delimiter = '.'): mixed
+    {
+        if (empty($path)) {
+            return null;
+        }
+        $path_array = explode($path_delimiter, $path);
+        $current = $array;
+        foreach ($path_array as $key) {
+            if (!isset($current[$key])) {
+                return null;
+            }
+            $current = $current[$key];
+        }
+        return $current;
+    }
+
+    /**
+     * Set an element within the array at the given path.
+     * @param array $array The array to check
+     * @param string $path The path to the element
+     * @param mixed $value The value to set
+     * @param string $path_delimiter The delimiter used in the path
+     * @return void
+     */
+    public static function setElementByArrayPath(array &$array, string $path, mixed $value, string $path_delimiter = '.'): void
+    {
+        if (empty($path)) {
+            return;
+        }
+        $path_array = explode($path_delimiter, $path);
+        $current = &$array;
+        foreach ($path_array as $key) {
+            if (!isset($current[$key])) {
+                $current[$key] = [];
+            }
+            $current = &$current[$key];
+        }
+        $current = $value;
+    }
+
+    /**
+     * Get all paths of an array matching a regex pattern
+     * @param array $array The array to check
+     * @param string $regex The regex pattern to match against the full paths
+     * @param string $path_delimiter The delimiter used in the path
+     * @return array
+     */
+    public static function getArrayPaths(array $array, string $regex = '/.*/', string $path_delimiter = '.'): array
+    {
+        // Get all paths including intermediate paths
+        $paths = [];
+        $iterator = new \RecursiveIteratorIterator(new \RecursiveArrayIterator($array), \RecursiveIteratorIterator::SELF_FIRST);
+        foreach ($iterator as $leafValue) {
+            $keys = [];
+            foreach (range(0, $iterator->getDepth()) as $depth) {
+                $keys[] = $iterator->getSubIterator($depth)->key();
+            }
+            // Implode keys to get all paths. For example, ['a', 'b', 'c'] will give ['a', 'a.b', 'a.b.c']
+            $temp = '';
+            foreach ($keys as $key) {
+                $temp .= $path_delimiter . $key;
+                $path = substr($temp, 1);
+                if (!empty($path) && preg_match($regex, $path) === 1) {
+                    $paths[] = $path;
+                }
+            }
+        }
+        $paths = array_unique($paths);
+
+        $t = '';
+        return $paths;
+    }
```

Comment:
Maybe we could use `symfony/property-access` (https://symfony.com/doc/current/components/property_access.html) for this. It would reduce the own code we have to maintain/test.

Reply:
This doesn't look like it would be a good solution for this use case.

When refering to locations in an array, you have to use index notation like `[key][sub_key]` and it is not configurable.
Dot notation like `key.sub_key` is only used for refering to object properties.

We only care about arrays and it must accept dot notation paths.
The API search system already denotes flattened property names in dot notation, and excluding arrays, lines up perfectly with the schema.
Example: `category.name` refers to the `name` property inside `category`.
```php
{
    "category": {
        "id": 1,
        "name": "example"
    }
}
```

Having to add logic to convert paths to index notation would add more complexity to an already fairly complex feature.
Further, the Symfony component doesn't seem to offer any replacement for `getArrayPaths` to get all possible paths for an array that match a pattern, so that would end up being maintained in GLPI anyways.

Reply:
Indeed, notation differences could be an issue.

Could you then move these methods into a new class `Glpi\Toolbox\ArrayAccessor` (or any better name) and add unit tests on it?